hc/front/views.py

```diff
@@ -27,6 +32,112 @@ def pairwise(iterable):
     return zip(a, b)
 
 
+def blogs(request, filter_by):
+    """function to render the home page"""
+    num = 3 # setting number of blog posts to be displayed as three per row
+    category = Category.objects.all()
+    if int(filter_by):
+        category_id = Category.objects.get(pk=filter_by).id
+        stories = Blog_post.objects.filter(category=category_id)
+    else:
+        stories = Blog_post.objects.all()
+    list_of_stories = [story for story in stories]
+    blogs = [list_of_stories[item:item+num] for item in range(0, len(list_of_stories), num)]
+    ctx = {
+        'category':category,
+        'blogs':blogs
+    }
+    return render(request, 'front/blog_posts.html', ctx)
+
+
+def create_blog(request):
+    """function to create a category and a blog """
+    form = CreateBlogPost(request.POST)
+    category_form = CreateCategory(request.POST)
+    if request.method == 'POST':
+        if 'new_category' in request.POST:
```

Comment:
Unnecessary nesting of `if` statements. Since the inner `if` has no else and it is the only one, you can conjoin that logic to the outer conditional statement. Something like this;

```
if 'new_category' in request.POST and 'category_form.is_valid():
  # your logic goes here
```